api/src/main/java/uk/gov/dvsa/mot/vehicle/hgv/HgvVehicleProvider.java

```diff
@@ -0,0 +1,143 @@
+package uk.gov.dvsa.mot.vehicle.hgv;
+
+import com.google.inject.Inject;
+
+import org.apache.log4j.Logger;
+import org.glassfish.jersey.apache.connector.ApacheConnectorProvider;
+import org.glassfish.jersey.client.ClientConfig;
+import org.glassfish.jersey.client.ClientProperties;
+import org.glassfish.jersey.client.JerseyClientBuilder;
+
+import uk.gov.dvsa.mot.vehicle.hgv.model.TestHistory;
+import uk.gov.dvsa.mot.vehicle.hgv.model.Vehicle;
+import uk.gov.dvsa.mot.vehicle.hgv.response.ResponseTestHistory;
+import uk.gov.dvsa.mot.vehicle.hgv.response.ResponseVehicle;
+
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.core.Response;
+
+public class HgvVehicleProvider {
+    private static final Logger logger = Logger.getLogger(HgvVehicleProvider.class);
+
+    private HgvConfiguration configuration;
+
+    @Inject
+    public void setHgvConfiguration(HgvConfiguration configuration) {
+        this.configuration = configuration;
+    }
+
+    public Vehicle getVehicle(String registration) throws Exception {
+        logger.trace("Entering HgvVehicleProvider getVehicle()");
+
+        if (registration == null || registration.isEmpty()) {
+            throw new IllegalArgumentException("Registration is null or empty");
+        }
+
+        Vehicle vehicle;
+        TestHistory[] vehicleTestHistory;
+
+        try {
+            CompletableFuture<Vehicle> vehicleFuture = CompletableFuture.supplyAsync(() -> {
+                Vehicle result = null;
+
+                try {
+                    result = getHgvVehicle(registration);
+                } catch (IOException e) {
+                    logger.error("IO error during communication with HGV vehicle api",  e);
```

Comment:
catch cascade is almost duplicated between vehicle and history... can we DRY it ?